### PR TITLE
Code improvements and minor fixes before v2.0.0

### PR DESCRIPTION
**Pull request by hizir-bc**

### CODE REVIEW

**Comment by hizir-bc** [comment]
/supportive-cases/cascade ✅
/supportive-cases/db2 ✅

**Comment by hizir-bc** [comment]
/supportive-cases/cascade ✅
/supportive-cases/db2 ✅

**Comment by hizir-bc** [comment]
/supportive-cases/zos-connect ✅

**Comment by hizir-bc** [comment]
/supportive-cases/ca-view ✅

**Comment by hizir-bc** [comment]
pull request contains:

- readme updates
- single object search functionality fix implementation (for actual code and mock)
- some minor improvements

**Comment by hizir-bc** [comment]
all the manual rests are green ✅

File: README.md [hunk 1]
```diff
@@ -45,8 +45,7 @@ Once you have met the prerequisites,
 
 ## Notes
  
-1. Exclude feature of the Compare endpoint requires a certain File Master Plus version 
-   12.0 and the required PTF installation. Make sure you have the required PTF to be able to use the field exclusion.
+1. Exclude feature of the Compare endpoint requires a certain File Master Plus version 12.0 and the required PTF installation. Make sure you have the required PTF to be able to use the field exclusion.
```
Comment (sergiuilie) [inline README.md:48]: adding @vahidbaharvand to check the text

**Review by sergiuilie — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (hizir-bc) [reply README.md:48]: @sergiuilie he composed the text, thanks

**Review by vahidbaharvand — APPROVED** [review]

**Pull request merged**